src/swap/defi/xrpDex.ts

```diff
@@ -0,0 +1,275 @@
+    const toCurrency =
+      toTokenId != null ? toTokenId.split('-')[0] : toCurrencyCode
+
+    // Grab addresses:
+    const toAddress = await getAddress(toWallet)
+
+    const now = Date.now()
+    if (
+      now - exchangeInfoLastUpdate > EXCHANGE_INFO_UPDATE_FREQ_MS ||
+      exchangeInfo == null
+    ) {
+      try {
+        const exchangeInfoResponse = await promiseWithTimeout(
+          fetchInfo(fetchCors, `v1/exchangeInfo/${appId}`)
+        )
+
+        if (exchangeInfoResponse.ok === true) {
+          exchangeInfo = asExchangeInfo(await exchangeInfoResponse.json())
+          exchangeInfoLastUpdate = now
+        } else {
+          // Error is ok. We just use defaults
+          log('Error getting info server exchangeInfo. Using defaults...')
+        }
+      } catch (e: any) {
+        log(
+          'Error getting info server exchangeInfo. Using defaults...',
+          e.message
+        )
+      }
+    }
+
+    // ----------------------------------------------------------------
+    // Get a quote
+    // ----------------------------------------------------------------
+
+    const client = await getXrplConnectedClient(rippleServers)
+
+    let quote: number
+    let fromNativeAmount: string
+    let toNativeAmount: string
+    const taker = await getAddress(fromWallet)
+
+    if (quoteFor === 'from') {
+      fromNativeAmount = nativeAmount
+      const exchangeAmount = await fromWallet.nativeToDenomination(
+        nativeAmount,
+        fromCurrencyCode
+      )
+
+      quote = await getSellQuote(
+        {
+          weSell: {
+            currency: fromCurrency,
+            issuer: fromIssuer
+          },
+          weSellAmountOfTokens: Number(exchangeAmount),
+          counterCurrency: {
+            currency: toCurrency,
+            issuer: toIssuer
+          },
+          taker
+        },
+        { client, showLogs: false }
+      )
+      quote = quote * (1 - volatilitySpread)
+      toNativeAmount = await toWallet.denominationToNative(
+        String(quote),
+        toCurrencyCode
+      )
+      toNativeAmount = round(toNativeAmount, 0)
+    } else {
+      toNativeAmount = nativeAmount
+      const exchangeAmount = await toWallet.nativeToDenomination(
+        nativeAmount,
+        toCurrencyCode
+      )
+
+      quote = await getBuyQuote(
+        {
+          weWant: {
+            currency: toCurrency,
+            issuer: toIssuer
+          },
+          weWantAmountOfToken: Number(exchangeAmount),
+          counterCurrency: {
+            currency: fromCurrency,
+            issuer: fromIssuer
+          },
+          taker
+        },
+        { client, showLogs: false }
+      )
+      quote = quote * (1 + volatilitySpread)
+      fromNativeAmount = await fromWallet.denominationToNative(
+        String(quote),
+        fromCurrencyCode
+      )
+      fromNativeAmount = round(fromNativeAmount, 0)
+    }
+
+    const timestampNow = new Date().getTime()
```

Comment:
`new Date().getTime()` is the same as `Date.now()`.

Reply:
fixed